torchvision/models/squeezenet.py

```diff
@@ -36,10 +34,10 @@ def forward(self, x):
 
 class SqueezeNet(nn.Module):
 
-    def __init__(self, version='1_0', num_classes=1000):
+    def __init__(self, arch, num_classes=1000):
```

Comment:
This is BC-breaking as well.

One way to address this would be to have both `version` and `arch`, and handle the case where `arch` is missing (old way).

Reply:
This would be a trouble only for users who are using `SqueezeNet()` function for their model and not the `squeezenet1_0` or `squeezenet1_1`, right?

Reply:
Yes, that's correct. How many users call into the class instead of the modules? I don't know, but I have myself several times in the past inherited from models like `ResNet` to perform customization

Reply:
Using the class instead the modules for customization seems to be quite convenient. In that case I will revert the changes made to the `version` and let the removal of FIXME be only present. 

I think most of the users who use Pytorch as a framework read its docs and not the source code of the model. This hides the fact that some of the models support customized model implementation using the class itself. Do you think that adding a new module for accessing the class with customised hyperparameter along with adding them to the docs could be beneficial?

Reply:
> Do you think that adding a new module for accessing the class with customised hyperparameter along with adding them to the docs could be beneficial?

In general, we pass `**kwargs` to the functions, so that the user only need to subclass the model class if we want to obtain something else out of it, e.g., returning intermediate activations of it.

I'm not clear if we should explicitly specify something else, could you clarify?

Reply:
Suppose someone wants to make a custom resnet model with `planes=[3, 7, 14, 2]`. They can do that in the following way: 

`model = ResNet(inplanes=BasicBlock, planes=[3, 7, 14, 2], **kwargs)`

But the above functionality is not mentioned in the [docs](https://pytorch.org/docs/stable/torchvision/models.html). This customisation is not available in SqueezeNets but is available in DenseNets and ResNets. So would it do good if we could somehow mention this in the docs?

Reply:
Sure, we could mention that in the docs.

But I'd assume that most users won't need this, because there will be no pre-trained models available for that particular configuration.
And for users who do want to try out a few different things, they will probably check the source code anyway, right? That's why having a simple to understand implementation is very important

Reply:
That does make sense! So should I close this PR?

Reply:
I'm closing this PR then, but thanks anyway!